packages/sitecore-jss-react/src/components/SitecoreContext.test.tsx

```diff
@@ -0,0 +1,48 @@
+import React from 'react';
+import { ComponentFactory } from './sharedTypes';
+import { expect } from 'chai';
+// const iterate = require('leakage').iterate;
+import { SitecoreContext, SitecoreContextFactory } from './SitecoreContext';
+import { withSitecoreContext } from '../enhancers/withSitecoreContext';
+const create = require('react-test-renderer').create;
+
+const TestNestedComponent: React.FC = (props: any) => (<div>{props.sitecoreContext && 'test'}</div>);
+const TestNestedComponentWithContext: React.FC = withSitecoreContext()(TestNestedComponent as any);
+const arrayOfComponents: React.Component[] = [];
+const mockComponentFactory: ComponentFactory = (name: string) => arrayOfComponents[name as any];
+const singleton = new SitecoreContextFactory();
+
+describe('SitecoreContext', () => {
+  // it('should not leak memory', () => {
+  //   iterate(() => {
+  //     const wrappedComponent = create(
+  //       <SitecoreContext componentFactory={mockComponentFactory} contextFactory={singleton}>
+  //         <TestNestedComponentWithContext />
+  //       </SitecoreContext>
+  //     );
+  //
+  //     singleton.setSitecoreContext({...singleton.context, myCustomProp: "asdfasdfasdfas"});
+  //
+  //     wrappedComponent.unmount();
+  //   }, {iterations:100});
+  // });
+
+  it('should unsubscribe from SitecoreContextFactory on unmount', () => {
+    const wrappedComponent = create(
+      <SitecoreContext componentFactory={mockComponentFactory} contextFactory={singleton}>
+        <TestNestedComponentWithContext />
+      </SitecoreContext>
+    );
+
+    const myCustomSubscriberContext: () => Object = () => ({});
```

Comment:
@M1r1k Please, define variables with more simple names,
change this line, also it's not required to define a type for this function, because typescript automatically defines it, when you are using primitive types such as Object, Number...
```
const subscribe = () => ({});
```